resolver-dns/src/main/java/io/netty/resolver/dns/DefaultDnsServerAddressStreamProvider.java

```diff
@@ -52,7 +52,12 @@ public final class DefaultDnsServerAddressStreamProvider implements DnsServerAdd
         if (!PlatformDependent.isAndroid()) {
             // Only try to use when not on Android as the classes not exists there:
             // See https://github.com/netty/netty/issues/8654
-            DirContextUtils.addNameServers(defaultNameServers, DNS_PORT);
+            if (!PlatformDependent.isWindows()) {
+                // /etc/resolv.conf exists on Linux + macOS, but not on Android
+                defaultNameServers.addAll(ResolvConf.system().getNameservers());
```

Comment:
Should we only fallback if `DirContextUtils` fails ?

Reply:
Not sure it's worth it, because it does the same thing under the covers.